src/lib/hooks/transactions/updater.tsx

```diff
@@ -58,6 +61,15 @@ export default function Updater({ pendingTransactions, onCheck, onReceipt }: Upd
         () =>
           provider.getTransactionReceipt(hash).then((receipt) => {
             if (receipt === null) {
+              if (account) {
+                provider.getTransactionCount(account).then((transactionCount) => {
+                  const tx = pendingTransactions[hash]
+                  if (tx.nonce && tx.nonce < transactionCount) {
```

Comment:
can you add a comment to make it clear to keep the tx.nonce check because old saved txs won't have a tx.nonce?